rules/msk_topic_config_test.go

```diff
@@ -721,13 +710,27 @@ resource "kafka_topic" "good topic" {
     "compression.type"      = "zstd"
   }
 }`,
-			expected: []*helper.Issue{},
-		},
-	} {
+		expected: []*helper.Issue{},
+	},
+}
+
+func Test_MSKTopicConfigRule(t *testing.T) {
+	rule := &MSKTopicConfigRule{}
+
+	var allTests []topicConfigTestCase
+	allTests = append(allTests, replicationFactorTests...)
+	allTests = append(allTests, compressionTypeTests...)
+	allTests = append(allTests, cleanupPolicyTests...)
+	allTests = append(allTests, deletePolicyRetentionTimeTests...)
+	allTests = append(allTests, deletePolicyTieredStorageTests...)
+	allTests = append(allTests, goodConfigTests...)
```

Comment:
random thought: I really like the iterators in Go 1.23 that would allow us to do this lazily (i.e. loop over all this sub-arrays without constructing a new one)

Reply:
Not sure about this one... I think it's clearer what's happening like this.
I think in the implementing Seq func we'd still to iterate through each of these slices, or to unify them and then iterate, just like in here

Reply:
> Not sure about this one... I think it's clearer what's happening like this. I think in the implementing Seq func we'd still to iterate through each of these slices, or to unify them and then iterate, just like in here

Unless there w as a library for that :wink: https://pkg.go.dev/github.com/matthewhughes934/go-itertools@v0.4.0/itertools#ChainSlices (not saying to use it, just something I had some fun playing around with recently)

Reply:
:) I'll leave it like this, if you don't mind, as I don't want to introduce a dependency from a repo of some shady dude